src/content/en/fundamentals/media/fast-playback-with-video-preload.md

```diff
@@ -0,0 +1,578 @@
+      sourceBuffer.appendBuffer(data);
+      // TODO: Fetch file_2.webm when user starts playing video.
+    })
+    .catch(error => {
+      // TODO: Show "Video is not available" message to user.
+    });
+  }
+
+  // This function is called if the first segment of video is not available.
+  function preloadFailed() {
+    // TODO: Show "Video is not available" message to user.
+  }
+</script>
+```
+
+Warning: For cross-origin resources, make sure your CORS headers are set
+properly because opaque responses retrieved with `fetch(videoFileUrl, { mode:
+'no-cors' })` are not allowed to feed any video or audio element.
+
+### Support
+
+Link preload is not supported in every browser yet. You may want to detect its
+availability with the snippet below.
+
+```
+<link rel="preload" as="video" href="https://cdn.com/file_1.webm"
+    onload="preloadFinished()" onerror="preloadFailed()">
+
+<video id="video" controls></video>
+
+<script>
+  if (!preloadVideoSupported()) {
+    // TODO: Fetch video from the network and load it.
+  }
+
+  function preloadFinished() {
+    // TODO: Load video.
+  }
+
+  function preloadFailed() {
+    // TODO: Show "Video is not available" message to user.
+  }
+
+  function preloadVideoSupported() {
+    const link = document.createElement('link');
+    link.as = 'video';
+    return (link.as === 'video');
+  }
+</script>
+```
+
+## Manual Buffering
+
+Before we dive into the [Cache API] and service workers, let's see how to
+manually buffer a video with MSE. The example below assumes that your web
+server supports HTTP Range requests but this would be pretty similar with
+segment files. Note that some middleware libraries such as [Google's Shaka
+Player], [JW Player], and [Video.js] are dedicated to handle this for you.
+
+```
+<video id="video" controls></video>
+
+<script>
+  const mediaSource = new MediaSource();
+  video.src = URL.createObjectURL(mediaSource);
+  mediaSource.addEventListener('sourceopen', sourceOpen, { once: true });
+
+  function sourceOpen() {
+    URL.revokeObjectURL(video.src);
+    const sourceBuffer = mediaSource.addSourceBuffer('video/webm; codecs="vp9"');
+
+    // Fetch beginning of the video by setting the Range HTTP request header.
+    fetch('file.webm', { headers: { range: 'bytes=0-567139' } })
+    .then(response => response.arrayBuffer())
+    .then(data => {
+      sourceBuffer.appendBuffer(data);
+      sourceBuffer.addEventListener('updateend', updateEnd, { once: true });
+    });
+  }
+
+  function updateEnd() {
+    // Video is now ready to play!
+    var bufferedSeconds = video.buffered.end(0) - video.buffered.start(0);
+    console.log(bufferedSeconds + ' seconds of video are ready to play!');
+
+    // Fetch the rest of the video when user starts playing video.
+    video.addEventListener('playing', fetchRemainder, { once: true });
+  }
+
+  function fetchRemainder() {
+    fetch('file.webm', { headers: { range: 'bytes=567140-1196488' } })
```

Comment:
+1, with the additional change to the comment in the caller
From:
// Fetch the rest of the video when user starts playing the video
To:
// Fetch the next segment of video when user starts playing the video

And in fetchNextSegment(), add a TODO comment indicating that further segment fetching and appending is elided for simplicity?